src/ampycloud/layer.py

```diff
@@ -233,14 +233,24 @@ def ncomp_from_gmm(vals: np.ndarray,
     else:
         raise AmpycloudError(f'Unknown scores: {scores}')
 
+    # Fix #119: at times, not all sub-layers may be populated by GaussianMixture.
+    # To avoid problems down the line, we shall boost the abics score of such cases to make sure
+    # they are NOT taken as the best model
+    for (n_id, n_val) in enumerate(ncomp):
```

Comment:
I'm not in favor of a retry for the following two reasons: first, what if the retry also fails ? How many random seeds should one try before giving up ? I see a risk here to loop over many times to find a "proper" seed, which is not ideal. Second, as far as I understand, this situation occurs only when we try to fit 3 Gaussians to a distribution that obviously is NOT comprised of three Gaussians. There's no need to force `GaussianMixture()` to give us three "populated" Gaussians to decide if they make sense or not: the simple fact of not give 3 populated Gaussian tells us directly that the option makes no sense - boosting the score is just a way to rule it out (and since the score itself is not used for anything else).

Reply:
Fully agree.